Rebus.AdoNet/Dialects/PostgreSqlDialect.cs

```diff
@@ -46,8 +46,18 @@ public PostgreSqlDialect()
 
 		#region Overrides
 		public override bool SupportsSelectForUpdate => true;
+		public override bool SupportsTryAdvisoryLockFunction => true;
 		public override string ParameterSelectForUpdate => "FOR UPDATE";
 
+		public virtual string ParameterSkipLocked => string.Empty;
+
+		// Version less than 9.1
+		public virtual Version DatabaseVersion => new Version("8.4");

```

Comment:
You need at least three clases here:
- PostgreSqlDialect => The one intended for v8.4, inheriting from Dialect.
- PostgreSql91Dialect => The one intended for v9.1+, inheriting from PostgreSqlDialect.
- PostgreSql95Dialect => The one intended for v9.5+, inheriting from PostgreSql91Dialect.

Each class should override/extend properties (or methods) related to new features supported on each newer version.

Also, each class should be able to auto-determine wether they are supported on a given database instance, by checking it's version.